tests/unit/test_read_config.py

```diff
@@ -238,12 +299,26 @@ def test_009_config_object(input_str, config_path):
         if env_var in config.keys():
             config.pop(env_var)
 
+    # parse the dates
+    config["end_date"] = process_date_string(config["end_date"])
+    config["start_date"] = process_date_string(config["start_date"])
+
     # load config object (performs all of the above steps)
-    wrf_config = load_wrf_config(config_path)
+    wrf_config = load_wrf_config(config_path_wrf_docker)
 
     assert config == asdict(wrf_config)
 
 
+def test_011_valid_CMAQ_NCI_config_file(config_path_cmaq_nci):
```

Comment:
Can we use [pytest-regressions](https://pytest-regressions.readthedocs.io/en/latest/index.html) to verify that the hydrated configuration doesn't change?

That would use the `data_regression` fixture similar to this example in the `openmethane` repo https://github.com/openmethane/openmethane/blob/ead28333c2cc268bba1ad41073a5903f030e0e0a/tests/unit/fourdvar/test_params.py#L132.

Happy to merge this and do that in a follow up MR as you have another MR stacked on top of this

Reply:
I can do it in the next pull request.